.github/workflows/build_main.yml

```diff
@@ -0,0 +1,29 @@
+name: build everything
+on:
+  push:
+    branches:
+      - main
```

Comment:
I would run this build on all branches except `R`, `python`, and `scala`.
This way the `feature` branches get built all the time